astropy/visualization/wcsaxes/tests/test_wcsapi.py

```diff
@@ -266,6 +268,73 @@ def test_coord_type_from_ctype(cube_wcs):
     assert coord_meta["format_unit"] == [u.one, u.one]
     assert coord_meta["wrap"] == [None, None]
 
+    myframe_mapping = {
+        "custom:pos.myframe.lon": {
+            "coord_wrap": 180.0 * u.deg,
+            "format_unit": u.arcsec,
+            "coord_type": "longitude",
+        },
+        "custom:pos.myframe.lat": {"format_unit": u.arcsec, "coord_type": "latitude"},
+    }
+
+    astropy.visualization.wcsaxes.wcsapi.CUSTOM_UCD_COORD_META_MAPPING.update(
+        myframe_mapping
+    )
+
+    wcs = WCS(naxis=2)
+    wcs.wcs.ctype = ["MFLN-TAN", "MFLT-TAN"]
+    wcs.wcs.crpix = [256.0] * 2
+    wcs.wcs.cdelt = [-0.05] * 2
+    wcs.wcs.crval = [50.0] * 2
+    wcs.wcs.set()
+
+    custom_mapping = {
+        "MFLN": "custom:pos.myframe.lon",
+        "MFLT": "custom:pos.myframe.lat",
+    }
+    with custom_ctype_to_ucd_mapping(custom_mapping):
+        _, coord_meta = transform_coord_meta_from_wcs(wcs, RectangularFrame)
+
+    assert coord_meta["type"] == ["longitude", "latitude"]
+    assert coord_meta["format_unit"] == [u.arcsec, u.arcsec]
+    assert coord_meta["wrap"] == [180 * u.deg, None]
+
+
+def test_custom_coord_type_from_ctype():
+    wcs = WCS(naxis=1)
+    wcs.wcs.ctype = ["eggs"]
+    wcs.wcs.cunit = ["deg"]
+
+    custom_mapping = {
+        "eggs": "custom:pos.eggs",
+    }
+    with custom_ctype_to_ucd_mapping(custom_mapping):
+        fig = plt.figure()
+        ax = fig.add_subplot(111, projection=wcs)
+        assert ax.coords["eggs"].coord_type == "scalar"
+        assert ax.coords["eggs"].coord_wrap == None
+        assert ax.coords["eggs"].get_format_unit() == u.deg
+
+        custom_meta = {
+            "pos.eggs": {
+                "coord_wrap": 360.0 * u.deg,
+                "format_unit": u.arcsec,
+                "coord_type": "longitude",
+            }
+        }
+        with custom_ucd_wcscoord_mapping(custom_meta):
+            ax = fig.add_subplot(111, projection=wcs)
+            ax.coords
+            assert ax.coords["eggs"].coord_type == "longitude"
+            assert ax.coords["eggs"].coord_wrap == 360 * u.deg
+            assert ax.coords["eggs"].get_format_unit() == u.arcsec
```

Comment:
It would be good if possible to add a test of two nested ``custom_ucd_wcscoord_mapping`` calls to make sure both apply at the same time.

Reply:
Added